src/EduManagementLab.Core/Services/CourseService.cs

```diff
@@ -40,6 +45,16 @@ public Course GetCourse(Guid id, bool includeMembershipUsers = false)
             }
             return course;
         }
+        public IEnumerable<Course.Membership> GetUserCourses(Guid userId)
+        {
+            var userCourses = _unitOfWork.Courses.GetUserCourses(userId);
+            if (userCourses == null)
+            {
+                throw new UserNotFoundException(userId);
```

Comment:
Should UserNotFoundException be thrown here?
Maybe the user exists but it has no courses.

You can throw that error if the user is missing, but you should return an empty collection if the user exists but it has no memberships.

Reply:
Would 'Enumerable.Empty<Course.Membership>()' work in this case?

Reply:
I think so.

Reply:
But! I dont think you need this null check. Now the metod in the repository can be null but the query will return a empty collection if the user is missing or if the user dont have any courses.

Remove the ? from the repository method and interface and just return in the service method.